bin/generate_freq_table.rb

```diff
@@ -0,0 +1,41 @@
+require "frequency_table"
+require "clusterable/ht_item"
+require "services"
+require "milemarker"
+
+f = FrequencyTable.new
+log = Services.logger
+marker = Milemarker.new(batch_size: 5000)
+
+#Services.holdings_db.loggers << Logger.new($stdout);
+log.info("#{$$}: starting freq table")
+filename = ARGV[0]
+fh = File.open(filename,"r")
+
+ARGF.each_line do |line|
+  fields = line.split("\t",-1)
+  # ignore things missing collection code?
```

Comment:
Move this to part of the query in bin/cost_report_chunk.sh ?

`WHERE ... AND hf.collection_code IS NOT NULL`